src/libs/PersonalDetailsUtils.ts

```diff
@@ -111,32 +111,30 @@ function getNewPersonalDetailsOnyxData(logins: string[], accountIDs: number[]) {
              * Cleanup the optimistic user to ensure it does not permanently persist.
              * This is done to prevent duplicate entries (upon success) since the BE will return other personal details with the correct account IDs.
              */
-            successData[accountID] = null;
+            personalDetailsCleanup[accountID] = null;
         }
     });
 
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.PERSONAL_DETAILS_LIST,
+            value: personalDetailsNew,
+        },
+    ];
+
+    const successData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.PERSONAL_DETAILS_LIST,
+            value: personalDetailsCleanup,
+        },
+    ];
+
     return {
-        optimisticData: [
-            {
-                onyxMethod: Onyx.METHOD.MERGE,
-                key: ONYXKEYS.PERSONAL_DETAILS_LIST,
-                value: optimisticData,
-            },
-        ],
-        successData: [
-            {
-                onyxMethod: Onyx.METHOD.MERGE,
-                key: ONYXKEYS.PERSONAL_DETAILS_LIST,
-                value: successData,
-            },
-        ],
-        failureData: [
-            {
-                onyxMethod: Onyx.METHOD.MERGE,
-                key: ONYXKEYS.PERSONAL_DETAILS_LIST,
-                value: failureData,
-            },
-        ],
+        optimisticData,
+        successData,
+        failureData: [],
```

Comment:
They are included to improve code readability since when using API.write we also separate the different data types into variables instead of hardcoding the `OnyxUpdate[]` functions straight into the methods.